build/dockerfiles/dev.Dockerfile

```diff
@@ -6,81 +6,49 @@
+RUN dnf -y install libsecret libX11-devel libxkbcommon \
+    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/libsecret-devel-0.18.6-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/AppStream/x86_64/os/Packages/libxkbfile-1.1.0-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/PowerTools/x86_64/os/Packages/libxkbfile-devel-1.1.0-1.el8.x86_64.rpm" \
+    "https://rpmfind.net/linux/centos/8-stream/BaseOS/x86_64/os/Packages/zsh-5.5.1-6.el8_1.2.x86_64.rpm"
 
-# Download kubectl
-RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && \
-    chmod +x kubectl && \
-    mv kubectl /mnt/rootfs/usr/bin/
+# cleanup dnf cache
+RUN dnf -y clean all --enablerepo='*'
 
-# setup home folder inside the new fs
-ENV HOME=/mnt/rootfs/home/che
-ENV NPM_CONFIG_PREFIX=/mnt/rootfs/home/che/.npm-global
-ENV PATH /mnt/rootfs/opt/nodejs/bin:/mnt/rootfs/home/che/.npm-global/bin/:$PATH
+# Create `/etc/containers/containers.conf` configuration file
+# to increase amount of opened files
+#
+# [containers]
+# default_ulimits = [
+#  "nofile=65535:65535",
+# ]
+
+RUN (echo '[containers]'; echo 'default_ulimits = ['; echo ' "nofile=65535:65535",'; echo ']') >> /etc/containers/containers.conf
+
+USER 10001
 
-# install yarn
-RUN npm install -g yarn
+ENV NODEJS_VERSION=18.16.1
 
 ENV ELECTRON_SKIP_BINARY_DOWNLOAD=1 \
     PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=0
- 
-RUN cat /mnt/rootfs/etc/passwd | sed s#root:x.*#root:x:\${USER_ID}:\${GROUP_ID}::\${HOME}:/bin/bash#g > /mnt/rootfs/home/che/.passwd.template \
-    && cat /mnt/rootfs/etc/group | sed s#root:x:0:#root:x:0:0,\${USER_ID}:#g > /mnt/rootfs/home/che/.group.template
 
-COPY --from=machine-exec --chown=0:0 /go/bin/che-machine-exec /mnt/rootfs/bin/machine-exec
-COPY --chmod=755 /build/scripts/entrypoint-dev.sh /mnt/rootfs/entrypoint.sh
-COPY --chmod=755 /build/conf/dev/.p10k.zsh /mnt/rootfs/home/che/.p10k.zsh
+RUN source $NVM_DIR/nvm.sh && \
+    nvm install v$NODEJS_VERSION && \
+    nvm alias default v$NODEJS_VERSION && \
+    nvm use v$NODEJS_VERSION && \
+    npm install --global npm@9.7.2 && \
+    npm install --global yarn@v1.22.19
 
-RUN mkdir -p /mnt/rootfs/projects
+ENV PATH=$NVM_DIR/versions/node/v$NODEJS_VERSION/bin:$PATH
 
-RUN for f in "/mnt/rootfs/home/che" "/mnt/rootfs/etc/passwd" "/mnt/rootfs/etc/group" "/mnt/rootfs/projects" ; do\
-           chgrp -R 0 ${f} && \
-           chmod -R g+rwX ${f}; \
-       done
+USER 0
 
-FROM scratch
-COPY --from=ubi-micro-build /mnt/rootfs/ /
-ENV HOME=/home/che
-ENV NPM_CONFIG_PREFIX=/home/che/.npm-global
-ENV PATH /opt/nodejs/bin:/home/che/.npm-global/bin/:$PATH
-ENTRYPOINT /entrypoint.sh
-WORKDIR /projects
-RUN wget https://github.com/robbyrussell/oh-my-zsh/raw/master/tools/install.sh -O - | zsh && \
```

Comment:
Let's keep the "Oh My Zsh" framework installed. It makes the terminal in dev container much more dev-friendly )

BTW, when I open a terminal to the `dev` container, `bash` is running by default, but not `zsh` anymore, like we currently have in quay.io/che-incubator/che-code-dev:insiders.

Reply:
Do we consider https://github.com/robbyrussell/oh-my-zsh as trusted source?

If we compare popularity of `bash` and `zsh`, I think `bash` is more popular. Anyway, I didn't touch the configuration of the default shell. And it was even not configured for dev image.

Reply:
Btw, when I navigate to `https://github.com/robbyrussell/oh-my-zsh` I have been forwarded to `https://github.com/ohmyzsh/ohmyzsh`

Reply:
> Do we consider https://github.com/robbyrussell/oh-my-zsh as trusted source?

Sure, it's a trusted source )
https://ohmyz.sh/
<img width="484" alt="image" src="https://github.com/che-incubator/che-code/assets/1636395/6135527f-246b-4d9c-828d-2c2d6fd4e326">

Reply:
> Btw, when I navigate to https://github.com/robbyrussell/oh-my-zsh I have been forwarded to https://github.com/ohmyzsh/ohmyzsh

Great, we can use the GitHub org link.

Reply:
I found it, `chsh -s $(which zsh) root` changes the default shell for the terminal

Reply:
The command `chsh -s $(which zsh) root` changes the default shell for `root`. To change it for `user` we need to change it in the entrypoint [like this](https://github.com/che-incubator/che-code/blob/7.74.x/build/scripts/entrypoint-dev.sh#L16)

```
    echo "${USER_NAME:-che}:x:$(id -u):0:Eclipse Che:${HOME}:/bin/zsh" >> /etc/passwd
```

And I thought. What if we leave bash as default shell?
I've played a bit with `zsh` and did not find big benefits in comparing with `bash`, except it looks like new year tree.

Reply:
> And I thought. What if we leave bash as default shell?

I don't mind. We just need to make sure that ZSH still works as it's currently do in `che-incubator/che-code-dev:insiders`. The default shell can be easily configured through the `VS Code Default Profile`.


> except it looks like new year tree

Try to revert back the `.p10k.zsh` config
https://github.com/che-incubator/che-code/pull/253/files#diff-b3867b14785bf05eb0c09d7e9868fad2ceb8fcab8c3a12deefac0d5a485fc7b0L54
It customizes zsh, so it should look as minimalistic as it is currently in `che-incubator/che-code-dev:insiders`

Reply:
Done